src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,316 @@
+            print(f"Canceling job {job_id} (try #{retry})...")
+            tf_common.run_nersc_command(
+                ['scancel', str(job_id)], self.job_dir, log_file_prefix='scancel', check_return_code=True)
+
+            # Get the job status to confirm the job is canceled
+            job_status = self._get_job_status_from_nersc(job_id)
+
+        if job_status in [JobStatus.RUNNING, JobStatus.PENDING]:
+            raise ValueError(f"Failed to cancel job {job_id}.")
+        else:
+            print(f"Job {job_id} is canceled.")
+
+    def __init__(self, kbase_collection, load_ver, tool, root_dir=loader_common_names.ROOT_DIR,
+                 destroy_old_job_dir=False):
+        """
+        Initialize the task manager.
+
+        :param kbase_collection: KBase collection identifier name (e.g. GTDB).
+        :param load_ver: collection load version (e.g. r207.kbase.1).
+        :param tool: tool name (e.g. gtdb_tk, checkm2, etc.)
+        :param root_dir: root directory of the collection project.
+                         Default is the ROOT_DIR defined in src/loaders/common/loader_common_names.py
+        :param destroy_old_job_dir: if True, remove contents of the old job directory.
+        """
+
+        self.kbase_collection = kbase_collection
+        self.load_ver = load_ver
+        self.tool = tool
+        self.root_dir = root_dir
+
+        # job directory is named as <kbase_collection>_<load_ver>_<tool>
+        self.job_dir = os.path.join(self.root_dir, TASKFARMER_JOB_DIR,
+                                    f'{self.kbase_collection}_{self.load_ver}_{self.tool}')
+
+        self._create_job_dir(destroy_old_job_dir=destroy_old_job_dir)
+        self._tasks_df = self._retrieve_all_tasks()
+
+    def get_latest_task(self):
+        """
+        Get the latest task information
+
+        Tasks are sorted by job_submit_time in descending order. The latest task is the first row.
+        """
+        return self._tasks_df.iloc[0].to_dict() if self._task_exists() else {}
+
+    def retrieve_latest_task_status(self):
+        """
+        Retrieve the latest task status.
+        """
+        if not self._task_exists():
+            raise ValueError(f"Task does not exist for {self.kbase_collection}, {self.load_ver}, {self.tool}")
+
+        latest_task = self.get_latest_task()
+        job_id = latest_task["job_id"]
+        job_status = self._get_job_status_from_nersc(job_id)
+
+        return job_status, job_id
+
+    def submit_job(self, source_data_dir):
+        """
+        Submit the job to slurm
+
+        Follow the steps in https://docs.nersc.gov/jobs/workflow/taskfarmer/#taskfarmer and generate all the necessary
+        files for the job submission (e.g. wrapper script, task list and batch script, etc.) before calling this function.
+        """
+
+        job_status, job_id = self.retrieve_latest_task_status()
+        if job_status in [JobStatus.RUNNING, JobStatus.PENDING]:
+            self._cancel_job(job_id)
```

Comment:
Hmm, I wouldn't do this unless a `force` param or something was sent to the method, rather throw an error by default

Reply:
👍